File: Data Structures & Algorithms/three-integer-sum/submission-2.js
class Solution {
    /**
     * @param {number[]} nums
     * @return {number[][]}
     */
    threeSum(nums) {
       const res = new Set();
       nums.sort((a,b) => a-b);
        const len = nums.length;
       for(let i=0;i<len;i++){
        for(let j=i+1;j<len;j++){
            for(let k=j+1;k<len;k++){
                if(nums[i]+nums[j]+nums[k]==0){
                    res.add(JSON.stringify(
                        [nums[i],nums[j],nums[k]]
                        ));
                }
            }
        }
       }
       return Array.from(res).map((item) => 
       JSON.parse(item));
    }
}

File: Data Structures & Algorithms/daily-temperatures/submission-2.cpp
class Solution {
public:
    vector<int> dailyTemperatures(vector<int>& temp) {
        int l=0;
        int r=1;
        vector<int> res(temp.size(),0); 
        while(r<temp.size()){
            // cout<<temp[l]<<temp[r]<<endl;
           if(l!=r && temp[l]<temp[r]){
            // cout<<temp[r]<<endl;
              res[l]= (r-l);
              r=l;
              l++;
           }else if(r==(temp.size()-1) && l<r){
               r=l;
               l++;
           }
            r++;
         }
         return res;
    }
};
